table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/AstyanaxTableDAO.java

```diff
@@ -1394,9 +1463,76 @@ public void clearStashTokenRangeSnapshot(String stashId) {
         _stashTableDao.clearTokenRanges(stashId);
     }
 
+    @Timed (name = "bv.emodb.table.AstyanaxTableDAO.listUnpublishedDatabusEvents", absolute = true)
+    @Override
+    public Iterator<UnpublishedDatabusEvent> listUnpublishedDatabusEvents(DateTime fromInclusive, DateTime toExclusive) {
+        checkArgument(fromInclusive != null, "fromInclusive date cannot be null");
+        checkArgument(toExclusive != null, "toExclusive date cannot be null");
+
+        LocalDate fromInclusiveUTC = fromInclusive.withZone(DateTimeZone.UTC).toLocalDate();
+        // if toExclusive date falls on the next day as per the UTC time, we need to fetch the records even with that date.
+        LocalDate toExclusiveUTC = toExclusive.withZone(DateTimeZone.UTC).toLocalDate();
+        LocalDate toInclusiveDateUTC = toExclusiveUTC;
+        if (toExclusive.toLocalDate().compareTo(toExclusiveUTC) == 0) {
+            toInclusiveDateUTC = toExclusive.minusDays(1).withZone(DateTimeZone.UTC).toLocalDate();
+        }
```

Comment:
The above doesn't work. Even though `toExclusive` and `toExclusiveUTC` are of type `LocalDate` and that type has time components they are always set to midnight.  So the above comparison _always_ is true and `toInclusiveUTC` is _always_ decreased.  Here's the sequence I used which broke it:

```
$  curl -s -XPUT -H "Content-Type: application/json"     "http://localhost:8080/sor/1/_table/review:testcustomer?options=placement:'ugc_global:ugc'&audit=comment:'initial+provisioning',host:aws-tools-02"     --data-binary '{"type":"review","client":"TestCustomer"}' -H 'X-BV-API-Key: <key>'
{"success":true}
$ curl -s -XDELETE -H "Content-Type: application/json"     "http://localhost:8080/sor/1/_table/review:testcustomer?options=placement:'ugc_global:ugc'&audit=comment:'initial+provisioning',host:aws-tools-02"  -H 'X-BV-API-Key: <key>'
{"success":true}
$ $ curl "localhost:8080/sor/1/_unpublishedevents" -s -H "X-BV-API-Key: <key>" | jq .
[
  {
    "table": "review:testcustomer",
    "date": "2018-04-06T16:41:31.190Z",
    "eventType": "DROP_TABLE"
  }
]
$  curl "localhost:8080/sor/1/_unpublishedevents?from=2018-04-06T16:00Z&to=2018-04-06T16:42Z" -s -H "X-BV-API-Key: local_admin" | jq .
[]
```

To fix this you need to use the original `fromExclusive` which still has a time component attached, such as:

```
        LocalDate toInclusiveDateUTC = toExclusive.withZone(DateTimeZone.UTC).toLocalDate();
        if (toExclusive.withZone(DateTimeZone.UTC).get(DateTimeFieldType.millisOfDay()) == 0) {
            toInclusiveDateUTC = toInclusiveDateUTC.minusDays(1);
        }
```